zwave_js_server/const/command_class/lock.py

```diff
@@ -128,3 +131,68 @@ class CodeSlotStatus(IntEnum):
     CommandClass.DOOR_LOCK: TARGET_MODE_PROPERTY,
     CommandClass.LOCK: LOCKED_PROPERTY,
 }
+
+
+class DoorLockCCConfigurationSetOptionsDataType(TypedDict, total=False):
+    """Door Lock CC Configuration Set command options."""
+
+    # https://github.com/zwave-js/node-zwave-js/blob/master/packages/cc/src/cc/DoorLockCC.ts#L1156
+    operationType: int  # required
+    lockTimeoutConfiguration: int
+    outsideHandlesCanOpenDoorConfiguration: list[bool]  # required when from device
+    insideHandlesCanOpenDoorConfiguration: list[bool]  # required when from device
+    autoRelockTime: int
+    holdAndReleaseTime: int
+    twistAssist: bool
+    blockToBlock: bool
+
+
+DEFAULT_HANDLE_CONFIGURATION = [True, True, True, True]
+
+
+@dataclass
+class DoorLockCCConfigurationSetOptions:
+    """Door Lock CC Configuration Set command options."""
+
+    endpoint: Endpoint
+    operation_type: OperationType
+    lock_timeout_configuration: int | None = None
+    outside_handles_can_open_door_configuration: list[bool] = field(
+        default_factory=list
+    )
+    inside_handles_can_open_door_configuration: list[bool] = field(default_factory=list)
+    auto_relock_time: int | None = None
+    hold_and_release_time: int | None = None
+    twist_assist: bool | None = None
+    block_to_block: bool | None = None
+
+    def __post_init__(self) -> None:
+        """Post initialization."""
+        for attr_name in (
+            "inside_handles_can_open_door_configuration",
+            "outside_handles_can_open_door_configuration",
+        ):
+            if not getattr(self, attr_name):
+                setattr(self, attr_name, DEFAULT_HANDLE_CONFIGURATION)
+
+    def to_dict(self) -> DoorLockCCConfigurationSetOptionsDataType:
+        """Convert command options to dict."""
+        data = DoorLockCCConfigurationSetOptionsDataType(
+            operationType=self.operation_type,
+            outsideHandlesCanOpenDoorConfiguration=(
+                self.outside_handles_can_open_door_configuration
+            ),
+            insideHandlesCanOpenDoorConfiguration=(
+                self.inside_handles_can_open_door_configuration
+            ),
```

Comment:
Not 100% sure about this, my python is very rusty.
If `self.outside_handles_can_open_door_configuration` is a tuple, doesn't this result in `outsideHandlesCanOpenDoorConfiguration` being a nested tuple?

Reply:
The parenthesis here is just to be able to write the statement on more than one line.

A tuple with a single item is written like this:

```py
is_tuple = (1, )

```
This is not a tuple:
```py
not_tuple = (1)
```